### DISCUSSION TITLE
Changes in 3.0 that decrease accessibility, are poorly designed, or are hard to migrate to.

### DISCUSSION THREAD

**Discussion by sveltarne** [saadeghi/daisyui#2468] [Ideas / feature requests]
I'm currently going through the update process from 2.0 to 3.0 and there's a couple of changes that make the move difficult, and/or complicates code.

1. The new 'dark' theme. While i like it, the old 'dark' theme was a great option for sites that just want one neutral theme. In order to go back, you have to make a custom theme and use a color picker on every type of color, then copy it over to the config file. This may also turn off new users who don't want a dark/light mode and just want a more neutral theme.
2. `<select>` elements with the `select` class should have their text bolded, like it was in version 2. The text is hard to see without being bolded because its typically a similar color to base-100, which is the background of a select element.
3. the button class should still work properly if it is on an a element. After upgrading, any `btn` class on an a element resulted in the text being at the top instead of centered. If you're using the `btn` class, you want it to look like a DaisyUI button, it's common sense. 
4. Buttons use base-200 now, instead of neutral. Buttons need contrast with the background, so a difference of 100 isn't enough. that kind of difference in design suggests to the user a different area in design, not a button. And even if you do add the class `btn-neutral` to every button that was previously uncolored, if you hover over it it'll get lighter instead of darker like on 2.0. This one can't be fixed with a class, unless you add a new custom color with the exact hex code as the old button on hover.

These issues, combined with things like extra spacing in buttons that combines with margins users have previously added mean that upgrading from 2.0 to 3.0 (and future versions) is way more complicated than it should be, given the simplicity and efficiency that DaisyUI stands for. You may have to modify every page file in your project, and chances are the lengths of those files aren't getting shorter.

While design changes and technical changes need to happen over time, the changes listed here lead to worse designs, or make the experience worse for developers. There are two types of solutions depending on the problem:

- Complete Reversal:
    - Select text being non-bold(Given the background is of a similar color) and buttons being base-200 are objectively bad for accessibility, as low contrast has been discouraged in almost everyone's guidelines, from apple's to YouTube's to your average GitHub project's. The way they were in 2.0 should be the default.
    - The `btn` class not centering text on `<a>` elements is a needless change that virtually everyone will want to revert. While the <button> element is better for accessibility, the option of using an `<a>` element should still be there.
- Optional Reversal:
    - The 'classic dark theme' should be an optional theme you can enable.

"But what if its more of an inconvenience to undo all the migration changes for 3.0?"
Generally, it'll be pretty easy to undo migration changes. For example, to preserve the color of your buttons in 3.0, you'd have to add btn-neutral`. To undo this, you can search for that phrase in your code editor and have all instances deleted.

I'd like to conclude this by saying that DaisyUI is still great, even with these issues. While anyone could negate all of these changes in a few hours, I think these solutions need to be instituted, in order to reflect the Intuitive, Accessible, and Beginner-Friendly component library that DaisyUI is.

**Comment by saadeghi** [comment]
Thank you for the detailed feedback.  

About the color changes for the built-in `dark` theme:  
The changes were made based on user feedbacks. The previous colors were not dark enough for using a website at night. People were asking for darker color contrast which makes it easier to use in a dark environment. This is actually better for the accessibility. It's important to consider different kinds of displays and making sure the dark background doesn't hurt the eye. Especially for the default dark theme.
However I always encourage people to modify theme colors the way they need. Modifying themes in daisyUI is truly easy and if someone prefers the `dark` theme from version 2.x, it's available here:  

https://unpkg.com/browse/daisyui@2.52.0/src/colors/themes.js

You can put the colors from the theme you want to your tailwind.config.js and it just works. ([docs](https://daisyui.com/docs/themes/#-4)

About `select`  
There's a bug in Safari about having a bold font in select and we had to change it to normal weight: https://github.com/saadeghi/daisyui/issues/2311  
As soon as the bug is fixed in Safari we can bring back the bold weight again. It's unfortunate but we can't do anything about it on our end when there's a big in browsers and we can only avoid it.

About button class  
Can you please share an example code? I don't see any issues in that case.

About button background colors  
In version 2.x we didn't have any way to have a light colored button on a light colored theme and there was a demand for it. On the other hand, there are modifier class names for the button that applies each color so it makes more sense to have a `btn-neutral` modifier class to apply neutral color just like we have `btn-primary` to apply primary color, instead of forcing the neutral color for all buttons.  
It's easy to change this behavior of you wish, either in tailwind.config.js file or CSS file or using a class name on the button.

About "extra margin on the buttons" or "text not being centered"  
Can you please share an example? There's no margin for buttons at all and by default buttons and also `<a>` tags are inline elements and the text is at the center unless you have other CSS styles that is changing that behavior.

I assure you, no changes in version 3 was based on my personal preference, rather it was based on issue reports, research, careful thinking and decisions based on what is the best solution possible to provide an easy, accessible and customizable design system.

Let me know if you have any questions.

**Reply by sveltarne** [reply]
Thanks for the reply, i didn't expect to get one from the creator of the project!

### As for the new dark theme: 
Sorry if I didn't clarify this well enough, but I think the new dark theme should be **the** default and I like it! However, if DaisyUI includes the option of all these similar-looking light themes like cupcake theme, bumblebee theme, emerald theme, acid theme plus about half a dozen more like that, it makes sense that DaisyUI should have classic dark as an option built-in.

I don't think the classic dark theme should be included because it's a better dark theme, as the new one is better in dark rooms. Rather, it should be included because its a great option for websites that don't want dark/light modes, and just want the beautiful gray aesthetic that the classic dark theme had. The name could be changed to something besides classic dark theme (dark gray, gray, **charcoal**, **slate**) to reflect this, but personally I would call it classic dark theme so that more 2.0 converts will know about it.

If you guys decide not to add the classic dark theme as an option alongside bumblebee, etc, I suggest adding the link you sent to this line in the changelog...
- The default `dark` theme is now darker and it looks more comfortable when using a website at night.    

...as i don't see how anyone would find that link without making a similar GitHub discussion or finding this one:

In this scenario, I also think this should be added into the breaking changes section, as in practice, the new dark theme represents a new focus on looking good in a dark room. As I described earlier, this may not be as aligned with some users' design choices as the classic theme was, so its important to give them the easy option of switching back.

### As for select:
This makes sense. While I think that users with vision issues being able to see the text is more important than an unusual font, I understand where this is coming from. While this is not a thread requesting new elements or classes, this could be a good time to introduce a special select element using DaisyUI's own item picker. That would fix this problem and allow users to customize the font, or add descriptions for items. Just a thought though.

### Button class on `a` elements:

Apologies, that was caused by some change in how the menu class works. whenever i removed the parent element with the menu class, it started working.

### Button Background Color:

I understand this design choice now, although I don't agree with it, but my issue about the hover behavior still being different from 2.0 after adding `btn-neutral` ties back into the theme issue.

It turns out this happened because I recreated the classic dark theme from the old DaisyUI website's theme picker, which didn't include the focus and content variations that the link you sent does. I've resolved it by copying the dark theme from that link, but as I said new users probably won't come across that link unless you add it in the changelog or fix the problem by adding classic dark as a theme.

One last note, you should show users how to change this behavior in the breaking changes log.

As for 'text not being centered', that was the issue I was referring to when I said the button class wasn't working properly on `a` elements.

Lastly, when I say 'extra margin on buttons', I'm referring to the gap of 0.5 the buttons now have, which adds on to the margins that users have previously added in 2.0 to achieve the same effect. I think this is a good choice, I was just highlighting how things like this **combined** with the issues I've outlined lead to a lot of confusion when updating.

**Comment by saadeghi** [comment]
You may like the new `dim` theme added in daisyUI 4. It's has similar background color to the old `dark` theme, with improvements.

**Comment by sveltarne** [comment]
It's a good theme, but if you ask me its not that similar to the classic theme.
![image](https://github.com/saadeghi/daisyui/assets/114943151/cefda83e-1310-420e-85fb-db39fd8e12ab)
